cmd/activator/main.go

```diff
@@ -199,12 +206,52 @@ func main() {
 		return revisionInformer.Lister().Revisions(revID.Namespace).Get(revID.Name)
 	}
 
+	activationResultGetter := func(revID activator.RevisionID) *activator.ActivationResult {
```

Comment:
Juggling all of these functions seems awkward, especially with cross-calling.  I wonder if we could encapsulate this into an object with methods for each of these which we could pass as: `object.Method` in place of anonymous functions with implicit capture?